### ISSUE TITLE
Support String In ToolResultBlock param? 🧌

### ISSUE THREAD

**Issue by tarunamasa** [anthropics/anthropic-sdk-python#548]
# Question: Is excluding `str` from `ToolResultBlockParam.content` expected behavior?

I noticed that the current implementation of `ToolResultBlockParam` in `src/anthropic/types/tool_result_block_param.py` doesn't allow for a string in the `content` field. However, the documentation seems to suggest that this should be possible.

From the [documentation](https://docs.anthropic.com/en/docs/build-with-claude/tool-use):

> content: The result of the tool, as a string (e.g. "content": "15 degrees") or list of nested content blocks (e.g. "content": [{"type": "text", "text": "15 degrees"}]). These content blocks can use the text or image types.
> ](content: The result of the tool, as a string (e.g. "content": "15 degrees") or list of nested content blocks (e.g. "content": [{"type": "text", "text": "15 degrees"}]). These content blocks can use the text or image types.)

Currently, the code defines `content` as:

```python
Content = Union[TextBlockParam, ImageBlockParam]


class ToolResultBlockParam(TypedDict, total=False):
    tool_use_id: Required[str]

    type: Required[Literal["tool_result"]]

    content: Iterable[Content] 

    is_error: bool

```

This means that `content` cannot be a string

I'm considering updating this to:

```python
Content = Union[TextBlockParam, ImageBlockParam]


class ToolResultBlockParam(TypedDict, total=False):
    tool_use_id: Required[str]

    type: Required[Literal["tool_result"]]

    content: Iterable[Content] | str

    is_error: bool
```

This would allow for all three cases mentioned in the documentation:
1. A string (e.g., "15 degrees")
2. A list of content blocks

Is this the intended behavior to exclude the string option? Or was it missed?

Corresponding PR #547

**Comment by tarunamasa** [comment]
Corresponding code snippet in "calculator" notebook
```python
from anthropic.types import ToolResultBlockParam, TextBlockParam


def process_tool_call(tool_name, tool_input):
    if tool_name == "calculator":
        return calculate(tool_input["expression"])
    if tool_name == "search":
        return "the capital of Ecuador is Quito"


def chat_with_claude(user_message):
    print(f"\n{'='*50}\nUser Message: {user_message}\n{'='*50}")

    message = client.messages.create(
        model=MODEL_NAME,
        max_tokens=4096,
        messages=[{"role": "user", "content": user_message}],
        tools=tools,
    )

    print(f"\nInitial Response:")
    print(f"Stop Reason: {message.stop_reason}")
    print(f"Content: {message.content}")

    if message.stop_reason == "tool_use":
        tool_use = next(block for block in message.content if block.type == "tool_use")
        tool_name = tool_use.name
        tool_input = tool_use.input

        print(f"\nTool Used: {tool_name}")
        print(f"Tool Input: {tool_input}")

        tool_result = process_tool_call(tool_name, tool_input)

        print(f"Tool Result: {tool_result}")
        if not tool_result:
            tool_result = "No result found"

        tool_result_block = ToolResultBlockParam(
            tool_use_id=tool_use.id,
            content=tool_result,
            type="tool_result",
        )
        response = client.messages.create(
            model=MODEL_NAME,
            max_tokens=4096,
            messages=[
                {"role": "user", "content": user_message},
                {"role": "assistant", "content": message.content},
                {
                    "role": "user",
                    "content": [
                        tool_result_block,
                    ],
                },
            ],
            tools=tools,
        )
    else:
        response = message

    final_response = next(
        (block.text for block in response.content if hasattr(block, "text")),
        None,
    )
    print(response.content)
    print(f"\nFinal Response: {final_response}")

    return final_response
```
OR 
```python
from anthropic.types import ToolResultBlockParam, TextBlockParam


def process_tool_call(tool_name, tool_input):
    if tool_name == "calculator":
        return calculate(tool_input["expression"])
    if tool_name == "search":
        return "the capital of Ecuador is Quito"


def chat_with_claude(user_message):
    print(f"\n{'='*50}\nUser Message: {user_message}\n{'='*50}")

    message = client.messages.create(
        model=MODEL_NAME,
        max_tokens=4096,
        messages=[{"role": "user", "content": user_message}],
        tools=tools,
    )

    print(f"\nInitial Response:")
    print(f"Stop Reason: {message.stop_reason}")
    print(f"Content: {message.content}")

    if message.stop_reason == "tool_use":
        tool_use = next(block for block in message.content if block.type == "tool_use")
        tool_name = tool_use.name
        tool_input = tool_use.input

        print(f"\nTool Used: {tool_name}")
        print(f"Tool Input: {tool_input}")

        tool_result = process_tool_call(tool_name, tool_input)

        print(f"Tool Result: {tool_result}")
        if not tool_result:
            tool_result = "No result found"

        tool_result_block = ToolResultBlockParam(
            tool_use_id=tool_use.id,
            content=[TextBlockParam(text=tool_result, type="text")],
            type="tool_result",
        )
        response = client.messages.create(
            model=MODEL_NAME,
            max_tokens=4096,
            messages=[
                {"role": "user", "content": user_message},
                {"role": "assistant", "content": message.content},
                {
                    "role": "user",
                    "content": [
                        tool_result_block,
                    ],
                },
            ],
            tools=tools,
        )
    else:
        response = message

    final_response = next(
        (block.text for block in response.content if hasattr(block, "text")),
        None,
    )
    print(response.content)
    print(f"\nFinal Response: {final_response}")

    return final_response
```

**Comment by aaron-lerner** [comment]
Thanks for flagging, we'll get this fixed through our standard release channels.

**Comment by RobertCraigie** [comment]
This has been fixed in `v0.29.1`!
